libraries/abstractions/transport/secure_sockets/transport_secure_sockets.c

```diff
@@ -0,0 +1,507 @@
+                                                0,
+                                                SOCKETS_SO_REQUIRE_TLS,
+                                                NULL,
+                                                0 );
+
+    if( TransportSocketStatus != ( int32_t ) SOCKETS_ERROR_NONE )
+    {
+        LogError( ( "Failed to set secured option for soocket." ) );
+    }
+
+    /* Set ALPN option. */
+    if( ( TransportSocketStatus == SOCKETS_ERROR_NONE ) && ( pSocketsConfig->pAlpnProtos != NULL ) )
+    {
+        TransportSocketStatus = SOCKETS_SetSockOpt( tcpSocket,
+                                                    0,
+                                                    SOCKETS_SO_ALPN_PROTOCOLS,
+                                                    ppcALPNProtos,
+                                                    sizeof( ppcALPNProtos ) / sizeof( ppcALPNProtos[ 0 ] ) );
+
+        if( TransportSocketStatus != ( int32_t ) SOCKETS_ERROR_NONE )
+        {
+            LogError( ( "Failed to set ALPN option socket." ) );
+        }
+    }
+
+    /* Set SNI option. */
+    if( ( TransportSocketStatus == SOCKETS_ERROR_NONE ) && ( pSocketsConfig->disableSni == false ) )
+    {
+        TransportSocketStatus = SOCKETS_SetSockOpt( tcpSocket,
+                                                    0,
+                                                    SOCKETS_SO_SERVER_NAME_INDICATION,
+                                                    pHostName,
+                                                    hostnameLength + 1 );
+
+        if( TransportSocketStatus != ( int32_t ) SOCKETS_ERROR_NONE )
+        {
+            LogError( ( "Failed to set SNI option for socket. TransportSocketStatus %d, hostnameLength  + 1 %d ",
+                        TransportSocketStatus, hostnameLength + 1 ) );
+        }
+    }
+
+    /* Set custom server certificate. */
+    if( ( TransportSocketStatus == SOCKETS_ERROR_NONE ) && ( pSocketsConfig->pRootCa != NULL ) )
+    {
+        TransportSocketStatus = SOCKETS_SetSockOpt( tcpSocket,
+                                                    0,
+                                                    SOCKETS_SO_TRUSTED_SERVER_CERTIFICATE,
+                                                    pSocketsConfig->pRootCa,
+                                                    pSocketsConfig->rootCaSize );
+
+        if( TransportSocketStatus != ( int32_t ) SOCKETS_ERROR_NONE )
+        {
+            LogError( ( "Failed to set server certificate option for socket." ) );
+        }
+    }
+
+    return TransportSocketStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t transportTimeoutSetup( Socket_t tcpSocket,
+                                      uint32_t sendTimeoutMs,
+                                      uint32_t recvTimeoutMs )
+{
+    TickType_t receiveTimeout = 0, sendTimeout = 0;
+    int32_t TransportSocketStatus = ( int32_t ) SOCKETS_ERROR_NONE;
```

Comment:
We use _camelCase_ naming convention for local variables
```suggestion
    int32_t transportSocketStatus = ( int32_t ) SOCKETS_ERROR_NONE;
```